### ISSUE TITLE
Properly Implement SoftMax

### ISSUE THREAD

**Issue by danzimm** [danzimm/Cerebrum#15]
SoftMax regression should only really be used on the last layer of the network, and should be used with the log likelihood cost function. I want this to be statically compiled differently than if it wasn't a softmax neural network (so we can't pass a softmax flag at runtime). My current thinking is to have a custom cost function called softmaxlogliklihood. Then it just ignores whatever activation function is passed in and uses its own softmax function.
